traffic_ops/traffic_ops_golang/cachegroupparameter/parameters.go

```
@@ -180,3 +185,120 @@ func (cgparam *TOCacheGroupParameter) Delete() (error, error, int) {
+		return
+	}
+	defer inf.Close()
+	output, err := GetAllCacheGroupParameters(inf.Tx.Tx)
+	if err != nil {
+		api.WriteRespAlertObj(w, r, tc.ErrorLevel, "querying cachegroupparameters with error: "+err.Error(), output)
+		return
+	}
+	api.WriteResp(w, r, output)
+}
+
+func GetAllCacheGroupParameters(tx *sql.Tx) (tc.CacheGroupParametersList, error) {
+	parameters := map[string]string{
+		"orderby": "cachegroup",
+	}
+	// Query Parameters to Database Query column mappings
+	// see the fields mapped in the SQL query
+	queryParamsToQueryCols := map[string]dbhelpers.WhereColumnInfo{
+		"cachegroup":  dbhelpers.WhereColumnInfo{"cachegroup", api.IsInt},
+		"lastUpdated": dbhelpers.WhereColumnInfo{"last_updated", nil},
+		"parameter":   dbhelpers.WhereColumnInfo{"parameter", api.IsInt},
+	}
+	where, orderBy, pagination, _, errs := dbhelpers.BuildWhereAndOrderByAndPagination(parameters, queryParamsToQueryCols)
+	if len(errs) > 0 {
+		return tc.CacheGroupParametersList{}, util.JoinErrs(errs)
+	}
+
+	query := selectAllQuery() + where + orderBy + pagination
+	rows, err := tx.Query(query)
+	if err != nil {
+		return tc.CacheGroupParametersList{}, errors.New("querying cachegroupParameters: " + err.Error())
+	}
+	defer rows.Close()
+
+	paramsList := tc.CacheGroupParametersList{}
+	params := []tc.CacheGroupParametersNullable{}
+	for rows.Next() {
+		var p tc.CacheGroupParametersNullable
+		if err = rows.Scan(&p.CacheGroup, &p.Parameter, &p.LastUpdated); err != nil {
+			return tc.CacheGroupParametersList{}, errors.New("scanning cachegroupParameters: " + err.Error())
+		}
+		params = append(params, p)
+	}
+	paramsList.CacheGroupParameters = params
+	return paramsList, nil
+}
+
+func AddCacheGroupParameters(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	data, _ := ioutil.ReadAll(r.Body)
```

should we ignore the error here?

---

no we shouldnt. updated